libs/application/templates/health-insurance/src/lib/answerValidators.ts

```diff
@@ -0,0 +1,132 @@
+import isNumber from 'lodash/isNumber'
+import {
+  Application,
+  AnswerValidator,
+  AnswerValidationError,
+  getValueViaPath,
+} from '@island.is/application/core'
+import { StatusTypes, Status, FormerInsurance, Applicant } from '../types'
+import { NO, YES } from '../constants'
+import {
+  requireConfirmationOfResidency,
+  requireWaitingPeriod,
+} from '../healthInsuranceUtils'
+
+const buildValidationError = (
+  path: string,
+  index?: number,
+): ((message: string, field?: string) => AnswerValidationError) => (
+  message,
+  field,
+) => {
+  if (field && isNumber(index)) {
+    return {
+      message,
+      path: `${path}[${index}].${field}`,
+    }
+  }
+
+  return {
+    message,
+    path,
+  }
+}
+
+const STATUS = 'status'
+const FORMER_INSURANCE = 'formerInsurance'
+
+// TODO: Add translation messages here
+export const answerValidators: Record<string, AnswerValidator> = {
+  [STATUS]: (newAnswer: unknown, application: Application) => {
+    const field = `${STATUS}.type`
+    const buildError = buildValidationError(field)
+    const status = newAnswer as Status
+
+    if (!Object.values(StatusTypes).includes(status.type)) {
+      return buildError('You must select one of the above', field)
+    }
+    if (
+      status.type === StatusTypes.STUDENT &&
+      !status.confirmationOfStudies.length
+    ) {
+      return buildError('Please attach a confirmation of studies below', field)
+    }
+
+    return undefined
+  },
+  [FORMER_INSURANCE]: (newAnswer: unknown, application: Application) => {
+    const formerInsurance = newAnswer as FormerInsurance
+    const applicant = getValueViaPath(
+      application.answers,
+      'applicant',
+    ) as Applicant
+
+    // Registration must be Yes / No
+    if (
+      formerInsurance.registration !== YES &&
+      formerInsurance.registration !== NO
+    ) {
+      const field = `${FORMER_INSURANCE}.registration`
+      const buildError = buildValidationError(field)
+      return buildError('You must select one of the above', field)
+    }
+
+    // Check that country is a string and not empty
```

Comment:
Hm realised that we aren't really checking if it is a string, only if it is empty. So we should update this comment